scipy/stats/tests/test_multicomp.py

```diff
@@ -162,7 +153,60 @@ def test_critical_values(
     samples_4 = [[3.8, 2.7, 4.0, 2.4], [2.8, 3.4, 3.7, 2.2, 2.0]]
     control_4 = [2.9, 3.0, 2.5, 2.6, 3.2]
     pvalue_4 = [0.5832, 0.9982]
+    # Statistic, alternative p-values, and CIs computed with R multcomp `glht`
+    p_4_twosided = [0.58317, 0.99819]
+    p_4_greater = [0.30225, 0.69115]
+    p_4_less = [0.91929, 0.65212]
     statistic_4 = [0.90875, -0.05007]
+    ci_4_twosided = [[-0.6898153448579, -1.0333456251632],
+                     [1.4598153448579, 0.9933456251632]]
+    ci_4_greater = [-0.5186459268412, -0.8719655502147 ]
+    ci_4_less = [1.2886459268412, 0.8319655502147]
+    pvalues_4 = dict(twosided=p_4_twosided, less=p_4_less, greater=p_4_greater)
+    cis_4 = dict(twosided=ci_4_twosided, less=ci_4_less, greater=ci_4_greater)
+    case_4 = dict(samples=samples_4, control=control_4, statistic=statistic_4,
+                  pvalues=pvalues_4, cis=cis_4)
```

Comment:
I think all the significant differences in examples correspond with cases in which the mean of the experimental group is *greater* than the mean of the control group. This may hide a bug that would be exposed when the mean of the experimental group is *less* than the mean of the control group. (This happened with `pearsonr`.) If so, the bug may be related to the use of `abs` in the allowance calculation.